addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/models/v2/request/Mode.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.evohome.internal.api.models.v2.request;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Request model for the mode
+ *
+ * @author Jasper van Zuijlen - Initial contribution
+ *
+ */
+public class Mode {
+
+    public Mode(String mode) {
+        systemMode = mode;
+        timeUntil = null;
+        permanent = true;
+    }
+
+    public Mode(String mode, int day, int month, int year) {
+        systemMode = mode;
+        timeUntil = String.format("%s-%s-%sT00:00:00Z", year, month, day);
+        permanent = false;
+    }
+
+    @SerializedName("SystemMode")
+    public String systemMode;
+
+    @SerializedName("TimeUntil")
```

Comment:
Is the API really that inconistent in starting with lower or uppercase or is GSON case insensitive?

Reply:
IDK, never tested it lowercase. I heavily rely on the python implementation - which in turn is sniffed from the official app -  and they use a Capital there. I can test to see if it works with lower as well, but I think this mimics the behavior of the official app better.